src/antsibull_nox/config.py

```diff
@@ -173,6 +175,63 @@ class SessionLicenseCheck(_BaseModel):
     license_check_extra_ignore_paths: list[str] = []
 
 
+class ExecutionEnvironmentConfig(_BaseModel):
+    """
+    Execution enviroment check session config.
+    """
+
+    name: str
+    description: t.Optional[str] = None
+    test_playbooks: list[str]
+    version: t.Literal[3] = 3
+    base_image_name: str = "registry.fedoraproject.org/fedora-toolbox:latest"
+    ansible_core_source: t.Literal["package_pip", "package_system"] = "package_pip"
+    ansible_core_package: str = "ansible-core"
+    ansible_runner_package: str = "ansible-runner"
+    system_packages: list[str] = []
+    python_packages: list[str] = []
+    python_interpreter_package: t.Optional[str] = None
+    python_path: t.Optional[str] = None
```

Comment:
How about simplifying this to:
```suggestion
    config: dict[str, t.Any] = {}
```
Then you can specify anything in the EE definition explicitly:
```toml
[[sessions.ee_check.execution_environments]]
name = "fedora-toolbox"
description = "Testing EE builds with the fedora toolbox"
test_playbooks = ["ee.yml"]
config.images.base_image.name = "registry.fedoraproject.org/fedora-toolbox:latest"
config.dependencies.ansible_core.package_pip = "ansible-core"
config.dependencies.ansible_runner.package_pip = "ansible-runner"
config.dependencies.system = [
    "git",
    "curl",
]
config.dependencies.python = [
    "jinja2",
    "pyyaml",
    "requests",
]
config.dependencies.python_interpreter.package_system = "python3"
config.dependencies.python_interpreter.python_path = "/usr/bin/python3"
```